src/libasr/pass/python_bind.cpp

```diff
@@ -0,0 +1,567 @@
+        type = ASRUtils::TYPE(ASR::make_UnsignedInteger_t(al, l, 1));
+        break;
+    case U32:
+        type = ASRUtils::TYPE(ASR::make_UnsignedInteger_t(al, l, 4));
+        break;
+    case U64:
+        type = ASRUtils::TYPE(ASR::make_UnsignedInteger_t(al, l, 8));
+        break;
+    case F32:
+        type = ASRUtils::TYPE(ASR::make_Real_t(al, l, 4));
+        break;
+    case F64:
+        type = ASRUtils::TYPE(ASR::make_Real_t(al, l, 8));
+        break;
+    case STR:
+        type = ASRUtils::TYPE(ASR::make_Character_t(al, l, 1, -2, nullptr));
+        break;
+    case PTR:
+        type = ASRUtils::TYPE(ASR::make_CPtr_t(al, l));
+        break;
+    case PTR_TO_PTR:
+        type = ASRUtils::TYPE(ASR::make_Pointer_t(al, l, ASRUtils::TYPE(ASR::make_CPtr_t(al, l))));
+        break;
+    }
+    LCOMPILERS_ASSERT(type);
+    ASR::symbol_t *v = ASR::down_cast<ASR::symbol_t>(ASR::make_Variable_t(al, l, current_scope, s.c_str(al), nullptr,
+                                                            0, intent, nullptr, nullptr, ASR::storage_typeType::Default,
+                                                            type, nullptr, ASR::abiType::BindC, ASR::Public,
+                                                            ASR::presenceType::Required, true)); // intent == ASRUtils::intent_in ? true : false
+    current_scope->add_symbol(n, v);
+    return ASRUtils::EXPR(ASR::make_Var_t(al, l, v));
+}
+
+void declare_functions(Allocator &al, std::vector<struct F> fns, SymbolTable *parent_scope,
+                                std::string header_name="") {
+    Location *l = al.make_new<Location>();
+    l->first = 0;
+    l->last = 0;
+
+    Str s;
+    char *c_header = nullptr;
+    if (header_name != "") {
+        s.from_str(al, header_name);
+        c_header = s.c_str(al);
+    }
+
+    for (auto i: fns) {
+        Vec<ASR::expr_t*> args;
+        args.reserve(al, i.args.size());
+        s.from_str(al, i.m_name);
+        SymbolTable *current_scope = al.make_new<SymbolTable>(parent_scope);
+        int c = 0;
+        for (auto j: i.args) {
+            args.push_back(al, type_enum_to_asr_expr(al, j, *l, i.m_name + std::to_string(++c), current_scope,
+                                    ASRUtils::intent_in));
+        }
+        ASR::expr_t *retval = type_enum_to_asr_expr(al, i.retvar, *l, "_lpython_return_variable", current_scope,
+                                    ASRUtils::intent_return_var);
+        char *fn_name = s.c_str(al);
+        ASR::asr_t *f = ASRUtils::make_Function_t_util(al, *l, current_scope, fn_name, nullptr, 0, args.p, args.n,
+                                    nullptr, 0, retval, ASR::abiType::BindC, ASR::accessType::Public,
+                                    ASR::deftypeType::Interface, nullptr, false, false, false, false, false, nullptr, 0,
+                                    false, false, false, c_header);
+
+        parent_scope->add_symbol(i.m_name, ASR::down_cast<ASR::symbol_t>(f));
+    }
+}
```

Comment:
I was thinking about moving these functions to `asr_utils.cpp`.
These are generic functions that can be used to create new `BindC` `Interface` function declarations.

Any thoughts? @certik @Shaikh-Ubaid.

Reply:
Sounds good to me. Also see/consider if `pass_utils` can be a better option.

Reply:
I feel `asr_utils.cpp` is the better option. In the future, if we need to generate function declarations for using any C function, we could just use these functions. Putting it in `asr_utils.cpp` will let us use it in the AST to ASR step also (if necessary).